src/main.ts

```diff
@@ -2357,7 +2392,11 @@ function generateFromPartial(ctx: Context, fullName: string, messageDesc: Descri
     const objectProperty = getPropertyAccessor("object", fieldName);
 
     const readSnippet = (from: string): Code => {
-      if ((isLong(field) || isLongValueType(field)) && options.forceLong === LongOption.LONG) {
+      if (
+        (isLong(field) || isLongValueType(field)) &&
+        options.forceLong === LongOption.LONG &&
+        !isJsTypeFieldOption(options, field)
```

Comment:
Just curious, but we don't have any other `isJsTypeFieldOption` checks in here, but I imagine maybe that's because the types will line-up (for anything not this forceLong case?) and so we don't need any special handling?

Reply:
Yeah, exactly that. These cases will fall through to the `isPrimitive` check below, and just output reading from the variable as is (without any special method calls or handling)